rotkehlchen/chain/bitcoin/bch/manager.py

```diff
@@ -47,12 +54,23 @@ def __init__(self, database: 'DBHandler') -> None:
             cache_key=DBCacheDynamic.LAST_BCH_TX_BLOCK,
             api_callbacks=[BtcApiCallback(
                 name='haskoin',
-                balances_fn=self._query_haskoin_balances,
-                has_transactions_fn=self._query_haskoin_has_transactions,
-                transactions_fn=self._query_haskoin_transactions,
-            )],  # TODO: add other APIs for fallbacks if haskoin goes down
+                balances_fn=lambda accounts: self._query_haskoin_balances(base_url=HASKOIN_BASE_URL, accounts=accounts),  # noqa: E501
+                has_transactions_fn=lambda accounts: self._query_haskoin_has_transactions(base_url=HASKOIN_BASE_URL, accounts=accounts),  # noqa: E501
+                transactions_fn=lambda accounts, options: self._query_haskoin_transactions(base_url=HASKOIN_BASE_URL, accounts=accounts, options=options),  # noqa: E501
+            ), BtcApiCallback(
+                name='blockchain.info haskoin-store',
+                balances_fn=lambda accounts: self._query_haskoin_balances(base_url=BLOCKCHAIN_INFO_HASKOIN_BASE_URL, accounts=accounts),  # noqa: E501
+                has_transactions_fn=lambda accounts: self._query_haskoin_has_transactions(base_url=BLOCKCHAIN_INFO_HASKOIN_BASE_URL, accounts=accounts),  # noqa: E501
+                transactions_fn=lambda accounts, options: self._query_haskoin_transactions(base_url=BLOCKCHAIN_INFO_HASKOIN_BASE_URL, accounts=accounts, options=options),  # noqa: E501
+            ), BtcApiCallback(
+                name='melroy',
+                balances_fn=lambda accounts: query_blockstream_like_balances(base_url=MELROY_BASE_URL, accounts=accounts),  # noqa: E501
+                has_transactions_fn=lambda accounts: query_blockstream_like_has_transactions(base_url=MELROY_BASE_URL, accounts=accounts),  # noqa: E501
+                transactions_fn=None,  # TODO: add transactions from melroy
+            )],
         )
         self.converted_addresses: dict[BTCAddress, BTCAddress] = {}
+        self.last_haskoin_health: dict[str, tuple[bool, Timestamp]] = {}
```

Comment:
Can you explain what this mapping is for and what its stucture is? Why does it need a per base url dict? How many are there? And what the bool and timestamp mean.